### ISSUE TITLE
How long does the token lives? Can we set expiry limit?

### ISSUE THREAD

**Issue by kidino** [apioo/fusio#186]
I have been browsing the Fusio admin panel and I can't find anything about this. So I guess that's it then.

How long does the token lives? 

Can we set expiry limit?

**Comment by chriskapp** [comment]
Hi, yes , you can set the expiry limit for the tokens at the `configuration.php` file i.e.:

```
    'fusio_expire_app'        => 'P2D', // your API
    'fusio_expire_backend'    => 'PT1H', // backend app
    'fusio_expire_consumer'   => 'PT1H', // developer app
```

**Comment by jsalazar10** [comment]
Is there a why to make that the consumer token never expire?

**Comment by chriskapp** [comment]
Hi, so the consumer token does expire after 1 hour. Then the user needs to login again since the token becomes invalid. But you can change this as shown in the config above.

Regarding the second question if you want to reinstall Fusio you need to run the install command again. But you should make the installation on an empty database.

**Comment by victoralicaya** [comment]
how to set the configuration file so that fusio will never expire

**Comment by kidino** [comment]
Maybe you can set it expire in 999 years

```    'fusio_expire_app'        => 'P999Y', // your API

**Comment by chriskapp** [comment]
Yes, the solution of @kidino is basically the only solution since Fusio does not support tokens without expiration. From a security perspective it is also not recommended to create a token which never expires.
